liblwgeom/lwgeom_geos.c

```diff
@@ -1712,3 +1712,93 @@ LWGEOM* lwgeom_delaunay_triangulation(const LWGEOM *lwgeom_in, double tolerance,
 	
 #endif /* POSTGIS_GEOS_VERSION < 34 */
 }
+
+static
+GEOSCoordSequence* lwgeom_get_geos_coordseq_2d(const LWGEOM* g, uint32_t num_points)
+{
+	uint32_t i = 0;
+	uint8_t num_dims = 2;
+	LWPOINTITERATOR* it;
+	GEOSCoordSequence* coords;
+	POINT4D tmp;
+
+	coords = GEOSCoordSeq_create(num_points, num_dims);
+	if (!coords)
+		return NULL;
+
+	it = lwpointiterator_create(g);
+	while(lwpointiterator_next(it, &tmp))
+	{
+		if(!GEOSCoordSeq_setX(coords, i, tmp.x) || !GEOSCoordSeq_setY(coords, i, tmp.y))
+		{
+			GEOSCoordSeq_destroy(coords);
+			lwpointiterator_destroy(it);
+			return NULL;
+		}
+		i++;
```

Comment:
It looks like this might write past allocated memory if num_points > number of points in the input geometry.
In other words, why are you accepting a "num_points" in input at all ?